File: data/src/main/java/com/github/anrimian/musicplayer/data/database/utils/DatabaseUtils.java
package com.github.anrimian.musicplayer.data.database.utils;

import static com.github.anrimian.musicplayer.domain.utils.TextUtils.isEmpty;

public class DatabaseUtils {

    public static String[] getSearchArgs(String arg, int count) {
        if (isEmpty(arg)) {
            arg = null;
        }

        String[] result = new String[count];
        for (int i = 0; i < count; i++) {
            if (i == 0) {
                result[i] = arg;
            } else {
                result[i] = arg == null? null : "%" + arg + "%";
            }
        }
        return result;
    }

    public static Object[] toArgs(Object arg, int times) {
        Object[] result = new Object[times];
        for (int i = 0; i < times; i++) {
            result[i] = arg;
        }
        return result;
    }
}

File: data/src/main/java/com/github/anrimian/musicplayer/data/database/utils/DatabaseUtils.kt
package com.github.anrimian.musicplayer.data.database.utils

object DatabaseUtils {

    fun getSearchArg(arg: String?): String? {
        return arg?.let { "%$it%" }
    }

    fun getSearchArgs(arg: String?, count: Int): Array<String?> {
        val verifiedArg = if (arg.isNullOrEmpty()) null else arg
        return Array(count) { i ->
            if (i == 0) {
                verifiedArg
            } else {
                getSearchArg(verifiedArg)
            }
        }
    }

}
